logs/querysample/querysample.go

```diff
@@ -54,12 +55,22 @@ func transformExplainJSONToQuerySample(logLine state.LogLine, explainText string
 }
 
 var autoExplainTextPlanDetailsRegexp = regexp.MustCompile(`^Query Text: (.+)\s+([\s\S]+)`)
+var herokuAutoExplainWithTabRegexp = regexp.MustCompile(`^Query Text: ([^\t]+)\t([\s\S]+)`)
 
 func transformExplainTextToQuerySample(logLine state.LogLine, explainText string, queryRuntimeMs float64) (state.PostgresQuerySample, error) {
 	explainParts := autoExplainTextPlanDetailsRegexp.FindStringSubmatch(explainText)
 	if len(explainParts) != 3 {
 		return state.PostgresQuerySample{}, fmt.Errorf("auto_explain output doesn't match expected format")
 	}
+	// If EXPLAIN output's first char is not a capital letter (e.g. not something like "Update on" or "Index Scan"),
+	// likely it's hitting the Heroku's newline break in "Query Text:" chunk
+	// Handle the separation of the query and the explain output text with the tab for these cases
+	explainOutputFirstChar := explainParts[2][0]
+	if cUtil.IsHeroku() && !(explainOutputFirstChar >= 'A' && explainOutputFirstChar <= 'Z') {
```

Comment:
Wonder if it's worth trying to avoid `IsHeroku()` conditional checks from a maintenance perspective?

If you always checked for the text being in the expected character range, could the Heroku check be dropped?

e.g. flipping to:
```
(explainOutputFirstChar <= 'A' && explainOutputFirstChar >= 'Z'
```

Reply:
Thanks for the comment!
Ah yeah, we can flip that condition, though it's a bit different from what you suggested, more like the following:

```golang
explainOutputFirstChar < 'A' || explainOutputFirstChar > 'Z'
```

Reads like "if the first char is smaller than A or bigger than Z" then "do the workaround". Though maybe I misunderstand your suggestion, let me know in that case.

Regarding `IsHeroku()`, I thought limiting to it was a good idea to minimize changes to the currently working flow. I'd say this workaround is a bit hacky so I wanted to limit it.
That said, definitely open to adjust this. Have you observed this actually happens out side of Heroku before?

Reply:
Sounds good. Take my comments with a grain of salt. I didn't realize the collector is open source, which is cool to see! I agree I didn't properly invert your original condition. Added a comment above about that.

I've only used PgAnalyze with Heroku PostgreSQL so far, although I'm working with another client that's considering PgAnalyze, and they're on AWS RDS PostgreSQL.

Thanks!

Reply:
Appreciate the input!

> I've only used PgAnalyze with Heroku PostgreSQL so far, although I'm working with another client that's considering PgAnalyze, and they're on AWS RDS PostgreSQL.

Nice! In general we recommend setting up auto_explain in JSON format so these customers (e.g. RDS) won't hit this issue (won't touch this part of code). This is a bit of limitation that Heroku doesn't support JSON format + Heroku logs shape is a bit awkward. I'll remove `IsHeroku()` if we notice this happens on other providers and/or feel more confident on this workaround.

Thanks!